### PR TITLE
A try to resolve the problem of COMException in DTE object

### PR DESCRIPTION
**Pull request by gandarez**
Accordingly to https://msdn.microsoft.com/en-us/library/ms228772(v=vs.110).aspx using DTE2 could fix this problem.
Another article https://msdn.microsoft.com/en-us/library/ee834473.aspx

### CODE REVIEW

**Comment by gandarez** [comment]
@alanhamlett I decided to hold the solution name instead of trying to get it from DTE object every time. What do you think?

File: WakaTimePackage.cs [hunk 1]
```diff
@@ -240,8 +256,11 @@ private static void SettingsPopup()
 
         private static string GetProjectName()
         {
-            var projectName = _objDte.Solution != null && !string.IsNullOrEmpty(_objDte.Solution.FullName) ? _objDte.Solution.FullName : null;
-            return !string.IsNullOrEmpty(projectName) ? Path.GetFileNameWithoutExtension(projectName) : null;
+            return !string.IsNullOrEmpty(_solutionName)
```
Comment (alanhamlett) [inline WakaTimePackage.cs]: This nested ternary is a little confusing.

File: WakaTimePackage.cs [hunk 2]
```diff
@@ -21,18 +22,20 @@ namespace WakaTime
...
     {
         #region Fields
-        private static string _version = string.Empty;        
+        private static string _version = string.Empty;
         private static string _editorVersion = string.Empty;
-        private static WakaTimeConfigFile _wakaTimeConfigFile;
+        private static WakaTimeConfigFile _wakaTimeConfigFile;        
 
-        private static DTE _objDte;
+        private static DTE2 _objDte;
```
Comment (alanhamlett) [inline WakaTimePackage.cs]: why rename it to `DTE2`?

Comment (alanhamlett) [inline WakaTimePackage.cs]: Nevermind. We should delete the `using EnvDTE` above now that we are not using it.

Comment (gandarez) [inline WakaTimePackage.cs]: @alanhamlett DTE2 is the new interface for DTE, you can't call GetService passing DTE2.

File: WakaTimePackage.cs (hunk 1 above)
Comment (gandarez) [inline WakaTimePackage.cs]: It's not confusing lol

**Comment by alanhamlett** [comment]
Thank you! I'm testing this now and will release a new version to the gallery today.

**Comment by gandarez** [comment]
Let's try it @alanhamlett

**Comment by pritianka** [comment]
You guys are awesome; thanks for fixing VS - one of our most important
plugins!
ᐧ

Priyanka Sharma
Co-founder, WakaTime <http://www.wakatime.com> - Quantify your coding
www.twitter.com/pritianka
650-796-7125

**Pull request merged**